java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.java

```diff
@@ -79,14 +79,17 @@ public ArrowRecordBatch(int length, List<ArrowFieldNode> nodes, List<ArrowBuf> b
       arrowBuffers.add(new ArrowBuffer(offset, size));
```

Comment:
if you are looking for a more substantial win, you might try passing in the initial capacity to arrowBuffers above.

Reply:
Thanks for the good suggestion.